globus_automate_client/cli/flows.py

```diff
@@ -161,7 +198,12 @@ def flow_deploy(
         input_schema_dict,
         validate_definition=validate,
     )
-    format_and_echo(result, verbose=verbose)
+
+    # Match up output format with input format
+    if input_format == FlowInputFormat.json:
+        format_and_echo(result, json.dumps, verbose=verbose)
+    elif input_format == FlowInputFormat.yaml:
+        format_and_echo(result, yaml.dump, verbose=verbose)
```

Comment:
I'm inclined to add the format detection into the `format_and_echo` function to help keep the logic simple at this top level. Similar to my previous comment, I think a trying to dump json then backing off to dumping as yaml may make the most sense. Since `result` will always be a GlobusHTTP response, we can treat it as we need to in `format_and_echo`.

Reply:
I'm glad you saw this, it reminded me to ask you about what to do here. So this code handles the output format without using an output format param, it just matches up the output to be the same type as the input. This makes sense to me but I guess it boils down to these two choices:
1. Use the same output format as output format
2. Add an output format that defaults to json (same as elsewhere)

What do you think?

Reply:
Sorry this may be coming a little late - I've seen other CLI that let you specify output format via a CLI switch. We want to get there at some point (especially as we eventually look to move into the main globus cli) and I think it's reasonable to only support JSON output until we're there.

For reference the main globus CLI has these output formats: `-F, --format [unix|json|text]`